New Integration - SRP Energy (#14800)
home-assistant/home-assistant.github.io: source/_integrations/srp_energy.markdown

---
title: "SRP Energy"
description: "How to integrate SRP Energy within Home Assistant."
ha_category:
  - Energy
ha_release: 0.119
ha_iot_class: Cloud Polling
---

> The `srp_energy` component shows information from Srp hourly energy usage report for their customers.

Reviewer: ```suggestion
The SRP Energy integration shows information from SRP hourly energy usage report for their customers.
```

> You need a Username, Password, and AccountId which you can create at [Srp](https://www.srpnet.com).

Reviewer: ```suggestion
You need a username, password, and account ID, which you can create at [SRP](https://www.srpnet.com).
```

## Configuration

> Add Srp Energy to your installation from the configuraiton integration.

Reviewer: ```suggestion
Add SRP Energy to your installation from the configuration integration.
```

> Navigate to `configuration`, then `integrations` and click `add`. Click `Srp Energy` to start the configuration. 
> After completing the configuration flow, the `srp_energy` integration will be available.

Reviewer: ```suggestion
Navigate to "Configuration", then "Integrations" and click `+` button in the bottom right. Select `SRP Energy` to start the configuration. After completing the configuration procedure, the SRP Energy integration will be available.
```

{% configuration %}
username:
  description: Your username for SRP.
  required: true
  type: string
password:
  description: Your password for SRP.
  required: true
  type: string
id:
  description: Your account id for SRP.
  required: true
  type: string
is_tou:
  description: Your account is time of use planP.
  required: false
  type: boolean
  default: false

> {% endconfiguration %}

Reviewer: This is a configuration block for YAML configuration and should be removed.

> Details about the API are available in the [SRP Energy Developers API documentation](https://srpenergy-api-client-python.readthedocs.io/en/latest/?badge=latest).

Reviewer: Why is this needed in the end-user documentation? What does the user get from this or need this?

Author: Is there a recommended way to document the fields of a config flow page?

Reviewer: We currently, generally do not document those.

We have a configuration_basic block, which is currently mainly used by Lovelace UI config items that we could consider using for this purpose as well. But right now, we don't do that yet.

---
title: "SRP Energy"
description: "How to integrate SRP Energy within Home Assistant."
ha_category:
  - Energy

> ha_release: 0.13

Reviewer 2: Update release to 0.119

ha_iot_class: Cloud Polling
---

The `srp_energy` component shows information from Srp hourly energy usage report for their customers.

You need a Username, Password, and AccountId which you can create at [Srp](https://www.srpnet.com).

## Configuration

> To add Srp Energy to your installation, add the following to your `configuration.yaml` file:

Reviewer 2: I don't think this is accurate? The integration uses a config flow, so you shouldn't need to add anything to configuration.yaml?

```yaml
# Example configuration.yaml entry
sensor:
  - platform: srp_energy
    username: YOUR_USERNAME
    password: YOUR_PASSWORD
    id: YOUR_ACCOUNT_ID
```

{% configuration %}
username:
  description: Your username for SRP.
  required: true
  type: string
password:
  description: Your password for SRP.
  required: true
  type: string
id:
  description: Your account id for SRP.
  required: true
  type: string

> {% endconfiguration %}

Reviewer 2: Missing information for `is_tou` / `Is Time of Use` setting from config flow

---
title: "SRP Energy"
description: "How to integrate SRP Energy within Home Assistant."

> logo: srp_energy.png

Reviewer 3: ```suggestion
```
You can remove this line 😉
